### PR TITLE
feature(whl): add multiprocess support

### PR DESCRIPTION
**Pull request by kxzxvbk**
Bugs or todos:
- This code might only work on well on Linux
- Do not support torch.compile(model)

### CODE REVIEW

File: fling/utils/mp_utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        self.num_proc = num_proc
+        self.task_name = task_name
+
+    def launch(self, clients, **kwargs) -> list:
+        tasks = [(client, kwargs) for client in clients]
+        with multiprocessing.Pool(self.num_proc) as pool:
+            # Use starmap to apply the worker function to every task
+            # Each task is a tuple that contains the task object and the arguments
+            if self.task_name == 'train':
+                results = pool.starmap(client_trainer, tasks)
```
Comment (XinghaoWu) [inline fling/utils/mp_utils.py]: After my tests and my research, I found that ’Multiprocess‘ uses ’fock‘ to create child processes that copy variables from the main process. Therefore, the ’tasks‘ you pass in here will be **deeply copied** by the main process. After the multi-process training is complete, the clients' model in the main process is not updated。
In this code framework, one solution is to return 'model.state_dict()' in the client's 'train' function and execute 'load_state_dict()' in the pipeline.  In my tests, it works.

**Review by XinghaoWu — CHANGES_REQUESTED** [review]
The bug about MultiPorcess.

**Comment by kxzxvbk** [comment]
PyTorch2.0 Benchmark: 
A100 GPU + 4 core CPU + MNIST + CNN + batchsize=32
- num_proc = 1, total time per round = 21.7s
- **num_proc = 2, total time per round = 14.9s**
- num_proc = 3, total time per round = 15.3s
- num_proc = 4, total time per round = 16.5s
- num_proc = 8, total time per round = 25.2s

A100 GPU + 4 core CPU + CIFAR10 + resnet8 + batchsize=32
- num_proc = 1, total time per round = 45.2s
- num_proc = 2, total time per round = 32.1s
- **num_proc = 3, total time per round = 30.1s**
- num_proc = 4, total time per round = 32.5s
- num_proc = 8, total time per round = 40.4s

File: fling/utils/mp_utils.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            else:
+                raise ValueError(f'Unrecognized task name: {self.task_name}')
+
+        # Retrieve the loggers and updated clients respectively.
+        loggers = [results[i][0] for i in range(len(results))]
+        new_clients = [results[i][1] for i in range(len(results))]
+
+        # Copy the attributes of new clients to original clients.
+        for new_client in new_clients:
+            for client in clients:
```
Comment (XinghaoWu) [inline fling/utils/mp_utils.py]: Here might be able to simplify. The results of 'starmap' execution are ordered.

**Review by XinghaoWu — COMMENTED** [review]
A suggestion to improve the efficiency.

**Pull request merged**